llvm/lib/Support/BLAKE3/blake3_avx2_x86-64_unix.S

```diff
@@ -2,7 +2,7 @@
 
 #include "llvm_blake3_prefix.h"
 
-#if defined(__ELF__) && defined(__linux__)
+#if defined(__ELF__) && (defined(__linux__) || defined(__FreeBSD__) || defined(__Fuchsia__))
```

Comment:
This can be defined for all ELF OSes except possibly Solaris which may use an alternative section syntax.

Reply:
The Solaris issue is not the different section syntax (which would only apply on SPARC if `clang` supported `/usr/bin/as` as external assembler, which it doesn't), but the fact that Solaris doesn't care about `.note.GNU-stack` at all (cf. `llvm/lib/MC/MCAsmInfoELF.cpp:MCAsmInfoELF::getNonexecutableStackSection`.

Reply:
I've now changed the guards to match `MCAsmInfoELF::getNonexecutableStackSection`.  While we are consistent here now, the guards in `BLAKE3` differ from every other instance in the LLVM codebase.

Reply:
Ah, I forgot: tested on `amd64-pc-freebsd14.0` and `amd64-pc-solaris2.11`.